src/main/java/graphql/execution/MergedField.java

```diff
@@ -155,7 +156,7 @@ private Builder(MergedField existing) {
         }
 
         public Builder fields(List<Field> fields) {
-            this.fields = fields;
+            this.fields.addAll(fields);
             return this;
```

Comment:
used only in tests, could be a breaking change.

Maybe this is the root cause of the additional copy that I've just removed in the constructor.